onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -209,14 +234,20 @@ def change_password(self, request, *args, **kwargs):  # noqa
         if new_password:
             if not lock_out:
                 if user_profile.user.check_password(current_password):
+                    data = {
+                        'username': user_profile.user.username
+                    }
                     metadata = user_profile.metadata or {}
                     metadata['last_password_edit'] = timezone.now().isoformat()
                     user_profile.user.set_password(new_password)
                     user_profile.metadata = metadata
                     user_profile.user.save()
                     user_profile.save()
+                    data.update(invalidate_and_regen_tokens(
+                        user=user_profile.user))
```

Comment:
Do we need to return the access and temp token at this point? Could on the UI side simply make the request to the API/user endpoint to retrieve the new information from there as it already does on login?

Reply:
The tokens that are in UI will already be invalidated so I don't think they can be used to make a request to the user endpoint unless the user logs out, no?

Reply:
Yes, not necessarily logging out but would need to log in with the new credentials. Will we be terminating the existing session and creating a new one? Essentially as you say, log out and log in a fresh?

Reply:
No much particular objection but simply wondering.

Should we update the documentation at https://api.ona.io/static/docs/profiles.html#change-authenticated-users-password to reflect that we also return the access and temp tokens?

Reply:
We don't necessarily have to return the access and temp token at this point, returning them just makes it easier for the API consumer to continue making requesting (Just removes the added step of requesting for a new access token).

UI's will have to change / modify their stored token values. They can decide to terminate a session or just update it's values.

Reply:
Yes, we should update the documentation. Thank you for bringing that up, let me make that change.

Reply:
I'd like to think we'll be terminating the existing session, right @DavisRayM ?